app/src/main/java/io/intrepid/contest/screens/entrysubmission/join/JoinActivity.java

```diff
@@ -47,11 +48,13 @@ protected void onSubmitButtonClicked() {
     @Override
     public void enableSubmitButton() {
         enterCodeSubmitButton.setEnabled(true);
+        enterCodeSubmitButton.setVisibility(View.VISIBLE);
```

Comment:
same thing here, we dont really need enable/disable if we're showing and hiding.